### PR TITLE
fix: Fixes generation of release notes to not rely on existing release

### PR DESCRIPTION
**Pull request by ausias-armesto**
The execution of the `generate-changelog.ts` script relies that the release must exist to get the release date. But in the `release.yaml` workflow the release does not exist yet. So we need to add an edge case and set `now` for that scenario.

### CODE REVIEW

File: scripts/generate-changelog.ts [hunk 1]
```diff
@@ -195,8 +200,9 @@ async function getVersionDate(
   } else if (/^v?\d+\.\d+\.\d+$/.test(`${version}`)) {
     log("DEBUG", `Getting version date from release tag`);
     const tag = `${version}`.replace(/^v/, "");
-    const release = (await ghApiCall(config, repo, `/releases/tags/${tag}`)) as GitHubRelease;
-    date = release.created_at;
+    const release = (await ghApiCall(config, repo, `/releases/tags/${tag}`, true)) as GitHubRelease | null;
+    // Release may not exist yet if this is the currentVersion being created in this workflow run.
+    date = release?.created_at ?? new Date().toISOString();
```
[the remark this answers is not in the document]
Comment (ausias-armesto) [reply scripts/generate-changelog.ts]: Not need it

[the remark this answers is not in the document]
Comment (ausias-armesto) [reply scripts/generate-changelog.ts]: Fixed

File: scripts/generate-changelog.ts [hunk 2]
```diff
@@ -143,6 +144,10 @@ async function ghApiCall(
         }
       }
 
+      if (response.status === 404 && allowNotFound) {
```
[the remark this answers is not in the document]
Comment (ausias-armesto) [reply scripts/generate-changelog.ts:148]: Fixed
